pkg/integrations/manager.go

```diff
@@ -476,6 +476,12 @@ func (m *Manager) WireAPI(r *mux.Router) {
 		handler := loadHandler(key)
 		handler.ServeHTTP(rw, r)
 	})
+
+	for _, process := range m.integrations {
+		for route, handler := range process.i.CustomHandlers() {
+			r.Handle(route, handler)
+		}
+	}
```

Comment:
This bit isn't going to work with /-/reload support. If the integration gets restarted due to a configuration change, the handler will still be pointing to the old instance.

Reply:
Refactored the code to work with agent reload